lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
@zacholauson I am curious, what actually `entity?` should check under the hood? Its intent is unclear.